File: ferreteria-yomara-web/src/app/data/interfaces/venta-response.ts
export interface VentaResponse {
  id: number;
  codigoOperacion: string;
  tipoComprobante: string;
  medioPago: string;
  total: number;

  estado: string;
  estadoVenta:string | null | undefined;
  fechaVenta: string;
  
  // Logística
  tipoEntrega: string; 
  direccionEntrega?: string;
  referenciaEntrega?: string;
  telefonoContacto?: string;
  fechaProgramada?: string;
  
  // Cliente
  clienteId: number;
  clienteDocumento: string;
  clienteNombre: string;
}

File: ferreteria-api/src/main/java/pe/idat/yomara/ferreteria_api/model/dto/response/VentaResponse.java
package pe.idat.yomara.ferreteria_api.model.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
public class VentaResponse {
    private Long id;
    private String codigoOperacion;
    private String tipoComprobante;
    private String medioPago;
    private BigDecimal subtotal;
    private BigDecimal descuentoTotal;
    private BigDecimal impuesto;
    private BigDecimal total;
    private String estado;
    private String estadoVenta; // El Enum (PENDIENTE_DESPACHO, ENTREGADO, etc.)

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime fechaVenta;

    //  CAMPOS DE LOGÍSTICA (Nuevos en el DTO)
    private String tipoEntrega;      // TIENDA o DOMICILIO
    private String direccionEntrega;
    private String evidenciaEntrega; // URL o Base64 de la foto

    private Long clienteId;
    private String clienteDocumento;
    private String clienteNombre;

    private Long vendedorId;
    private String vendedorNombre;

    private Long sedeId;
    private String sedeNombre;

    private List<VentaDetalleResponse> detalles;
}
